src/BufferStream.ts

```diff
@@ -226,14 +227,19 @@ export class BufferStream {
 
     readHex(length) {
         var hexString = '';
-        for (var i = 0;i < length;i++) {
-            hexString += this.readUint8().toString(16).padStart(2, '0');
+        for (var i = 0; i < length; i++) {
+            let hexChar = this.readUint8().toString(16)
+            if (hexChar.length < 2) {
+                hexChar = '0' + hexChar;
+            }
+            hexString += hexChar;
         }
-        return hexString;
+        return hexString.toUpperCase();
     }
 
+
     checkSize(step) {
-        if (this.offset + step > this.buffer.byteLength) {
+        while (this.offset + step > this.buffer.byteLength) {
```

Comment:
This should mean that the buffer will still expand correctly if the step is larger than the current size